src/chainstate/stacks/boot/bns.clar

```diff
@@ -0,0 +1,973 @@
+;;;; Errors
+(define-constant ERR_PANIC 0)
+(define-constant ERR_NAMESPACE_PREORDER_NOT_FOUND 1001)
+(define-constant ERR_NAMESPACE_PREORDER_EXPIRED 1002)
+(define-constant ERR_NAMESPACE_PREORDER_ALREADY_EXISTS 1003)
+(define-constant ERR_NAMESPACE_UNAVAILABLE 1004)
+(define-constant ERR_NAMESPACE_NOT_FOUND 1005)
+(define-constant ERR_NAMESPACE_ALREADY_EXISTS 1006)
+(define-constant ERR_NAMESPACE_NOT_LAUNCHED 1007)
+(define-constant ERR_NAMESPACE_PRICE_FUNCTION_INVALID 1008)
+(define-constant ERR_NAMESPACE_PREORDER_CLAIMABILITY_EXPIRED 1009)
+(define-constant ERR_NAMESPACE_PREORDER_LAUNCHABILITY_EXPIRED 1010)
+(define-constant ERR_NAMESPACE_OPERATION_UNAUTHORIZED 1011)
+(define-constant ERR_NAMESPACE_STX_BURNT_INSUFFICIENT 1012)
+(define-constant ERR_NAMESPACE_BLANK 1013)
+(define-constant ERR_NAMESPACE_ALREADY_LAUNCHED 1014)
+(define-constant ERR_NAMESPACE_HASH_MALFORMED 1015)
+(define-constant ERR_NAMESPACE_CHARSET_INVALID 1016)
+
+(define-constant ERR_NAME_PREORDER_NOT_FOUND 2001)
+(define-constant ERR_NAME_PREORDER_EXPIRED 2002)
+(define-constant ERR_NAME_PREORDER_FUNDS_INSUFFICIENT 2003)
+(define-constant ERR_NAME_UNAVAILABLE 2004)
+(define-constant ERR_NAME_OPERATION_UNAUTHORIZED 2006)
+(define-constant ERR_NAME_STX_BURNT_INSUFFICIENT 2007)
+(define-constant ERR_NAME_EXPIRED 2008)
+(define-constant ERR_NAME_GRACE_PERIOD 2009)
+(define-constant ERR_NAME_BLANK 2010)
+(define-constant ERR_NAME_ALREADY_CLAIMED 2011)
+(define-constant ERR_NAME_CLAIMABILITY_EXPIRED 2012)
+(define-constant ERR_NAME_NOT_FOUND 2013)
+(define-constant ERR_NAME_REVOKED 2014)
+(define-constant ERR_NAME_TRANSFER_FAILED 2015)
+(define-constant ERR_NAME_PREORDER_ALREADY_EXISTS 2016)
+(define-constant ERR_NAME_HASH_MALFORMED 2017)
+(define-constant ERR_NAME_PREORDERED_BEFORE_NAMESPACE_LAUNCH 2018)
+(define-constant ERR_NAME_NOT_RESOLVABLE 2019)
+(define-constant ERR_NAME_COULD_NOT_BE_MINTED 2020)
+(define-constant ERR_NAME_COULD_NOT_BE_TRANSFERED 2021)
+(define-constant ERR_NAME_CHARSET_INVALID 2022)
+
+(define-constant ERR_PRINCIPAL_ALREADY_ASSOCIATED 3001)
+(define-constant ERR_INSUFFICIENT_FUNDS 4001)
+
+;;;; Constants
+(define-constant BURN_ADDRESS 'S0000000000000000000002AA028H)
+
+;; TTL
+;; todo(ludo): add real-life values
```

Comment:
We currently use a trick with the PoX contract to instantiate different constants based on whether or not we're running a test.  Want to employ that trick here too?

Reply:
I think we can keep the same constants for testnet and mainnet (and avoid any tricks), this todo is for getting the final values in place.